Block/Configuration.php

```diff
@@ -88,9 +90,13 @@ public function getConfiguration()
 
             if ($category && $category->getDisplayMode() !== 'PAGE') {
                 $category->getUrlInstance()->setStore($this->getStoreId());
-
+                $childCategories = $category->getChildrenCategories();
+                foreach ($childCategories as $childCategory){
```

Comment:
Unfortunately this will not handle nested categories - a recursive approach is needed.